### PR TITLE
Add missing metricsets

### PR DESCRIPTION
**Pull request by kvalliyurnatt**
This PR adds some of the missing metricsets to the stack monitoring configuration
fixes: #7277

### CODE REVIEW

**Review by naemono — COMMENTED** [review]
I think it's good, but we may want to note if there's a version requirement, see: https://github.com/elastic/cloud-on-k8s/issues/7277#issuecomment-1786802076

Also, run the full e2e test suite on this to make sure all versions are ok with this change.... I suspect older versions may fail.

**Comment by naemono** [comment]
buildkite test this -f p=gke -m s=7.17.8,s=8.15.0,s=8.16.0-SNAPSHOT

**Comment by naemono** [comment]
> Also, run the full e2e test suite on this to make sure all versions are ok with this change.... I suspect older versions may fail.

Update: I kicked them off for you....

**Comment by pebrc** [comment]
Yes I think we need a version gate for this.

**Comment by pebrc** [comment]
Just to capture a discussion we had out of band: the change required here needs to amend [the Metricbeat template](https://github.com/elastic/cloud-on-k8s/blob/main/pkg/controller/elasticsearch/stackmon/metricbeat.tpl.yml) for the ECK stack monitoring feature. We can also add it to the recipe as done here but the template is the main part and we can use conditionals etc there to make it version dependent

**Comment by kvalliyurnatt** [comment]
Added a version gate for the ingest pipeline and also added a unit test for it

**Comment by kvalliyurnatt** [comment]
buildkite test this -f p=gke -m s=7.17.8,s=8.15.0,s=8.16.0-SNAPSHOT

**Comment by kvalliyurnatt** [comment]
I tested this locally on my dev cluster
Deployed an Elasticsearch version of 8.3.3
```
metricbeat:
  modules:
    - hosts:
        - https://localhost:9200/
      metricsets:
        - ccr
        - cluster_stats
        - enrich
        - index
        - index_recovery
        - index_summary
        - ml_job
        - node
        - node_stats
        - pending_tasks
        - shard
        - pending_tasks
      module: elasticsearch
      password: VoW9n21635S61KbhuAEx07Xt
      period: 10s
      ssl:
        certificate_authorities:
          - /mnt/elastic-internal/kb-monitoring/default/source/certs/ca.crt
        enabled: true
        verification_mode: certificate
      username: elastic-internal-monitoring
      xpack:
        enabled: true
output:
  elasticsearch:
    hosts:
      - https://destination-es-http.default.svc:9200/
    password: d1L39L80A2WHSr7Q6Fjx0r5o
    ssl:
      certificate_authorities:
        - /mnt/elastic-internal/es-monitoring-association/default/destination/certs/ca.crt
      verification_mode: certificate
    username: default-source-default-destination-beat-es-mon-user
processors:
  - add_cloud_metadata: null
  - add_host_metadata: null
  ```
  no ingest pipeline in the metric set

deployed Elasticsearch 8.7.0, ingest pipeline present in the metricset
```
metricbeat:
  modules:
    - hosts:
        - https://localhost:9200/
      metricsets:
        - ccr
        - cluster_stats
        - enrich
        - index
        - index_recovery
        - index_summary
        - ml_job
        - node
        - node_stats
        - pending_tasks
        - shard
        - ingest_pipeline
        - pending_tasks
      module: elasticsearch
      password: c73262gARGsdO79GL421nYRV
      period: 10s
      ssl:
        certificate_authorities:
          - /mnt/elastic-internal/kb-monitoring/default/source/certs/ca.crt
        enabled: true
        verification_mode: certificate
      username: elastic-internal-monitoring
      xpack:
        enabled: true
output:
  elasticsearch:
    hosts:
      - https://destination-es-http.default.svc:9200/
    password: Dr2gwA3171o2qQL5rTMpf697
    ssl:
      certificate_authorities:
        - /mnt/elastic-internal/es-monitoring-association/default/destination/certs/ca.crt
      verification_mode: certificate
    username: default-source-default-destination-beat-es-mon-user
processors:
  - add_cloud_metadata: null
  - add_host_metadata: null
  ```

**Review by naemono — APPROVED** [review]
looks good to me.

File: pkg/controller/elasticsearch/stackmon/metricbeat.tpl.yml [hunk 1]
```diff
@@ -9,9 +9,14 @@ metricbeat.modules:
       - index_recovery
       - index_summary
       - ml_job
+      - node
```
Comment (thbkrkr) [inline pkg/controller/elasticsearch/stackmon/metricbeat.tpl.yml:12]: Oh!

File: pkg/controller/elasticsearch/stackmon/metricbeat.tpl.yml [hunk 2]
```diff
@@ -9,9 +9,14 @@ metricbeat.modules:
...
       - node_stats
       - pending_tasks
       - shard
+      {{- if isVersionGTE "8.7.0" }}
+      - ingest_pipeline
+      {{- end }}
+
```
Comment (thbkrkr) [inline pkg/controller/elasticsearch/stackmon/metricbeat.tpl.yml:19]: Same nit: keep alphabetical order?

**Review by thbkrkr — APPROVED** [review]
LGTM

File: pkg/controller/elasticsearch/stackmon/metricbeat.tpl.yml (hunk 2 above)
Comment (kvalliyurnatt) [reply pkg/controller/elasticsearch/stackmon/metricbeat.tpl.yml]: done

**Pull request merged**